accounts-db/src/ancient_append_vecs.rs

```diff
@@ -156,13 +156,16 @@ impl AncientSlotInfos {
         let threshold_bytes = self.total_alive_bytes_shrink.0 * percent_of_alive_shrunk_data / 100;
         for info_index in &self.shrink_indexes {
             let info = &mut self.all_infos[*info_index];
+            if bytes_to_shrink_due_to_ratio.0 < threshold_bytes {
+                // adding here means we won't include the storage that would exceed the threshold
+                bytes_to_shrink_due_to_ratio += info.alive_bytes;
```

Comment:
add first, so that we skip the storage that causes us to exceed the limit. This prevents a large storage from blowing the entire budget for packing.